src/uproot/behaviors/TH1.py

```diff
@@ -190,9 +191,73 @@ def to_boost(self, metadata=boost_metadata, axis_metadata=boost_axis_metadata):
 
         Converts the histogram into a ``boost-histogram`` object.
         """
-        raise NotImplementedError(repr(self))
+        if axis_metadata is None:
+            axis_metadata = boost_axis_metadata
+        if metadata is None:
+            metadata = boost_metadata
+
+        boost_histogram = uproot.extras.boost_histogram()
+
+        values = self.values(flow=True)
+
+        # 'fSumw2' will never be missing, if weights are not defined it is an array of length 0
+        sumw2 = None
+        if self.weighted:
```

Comment:
The `self.weighted` property actually checks for `nullptr`, as well as `fSumw2` just not being defined in the TStreamerInfo (which I don't think has been the case in any 21st century version of ROOT).

https://github.com/scikit-hep/uproot5/blob/10d87e242c13a16131875563863369ae4b253433/src/uproot/behaviors/TH1.py#L230-L233

```suggestion
        sumw2 = None
        if self.weighted:   # ensures self.member("fSumw2") exists
```

Reply:
I just noticed that you've reimplemented the `weighted` property, but I set the implementation back to the old definition.

It would be possible to prove that checking for `fSumw2 == nullptr` is unnecessary by testing files made by a suite of old ROOT versions. That's something that I did at the beginning of Uproot's development: I selected popular ROOT versions, which had been released approximately once every 6 months, going back 5 years before the start of Uproot development, which was 2017. Then I installed each on a virtual machine and generated data in each: TTrees and (I _think_!) histograms, then checked what each of these generated. I don't remember very well the exact outcome of that study or where to find it, and if I've misremembered, I might be following a superstition to still be checking for `fSumw2 == nullptr`.

The study could be repeated for a set of ROOT versions we care about today, but it hardly seems worth it; we can just be a little overcautious, especially when it's wrapped up in a function call.